### PR TITLE
[CPU][Runtime] Set some CPURuntime funcs visibility to default

### PR DESCRIPTION
**Pull request by niuxiaog**
To solve issue #293.

### CODE REVIEW

File: lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp [hunk 1]
```diff
@@ -239,16 +239,24 @@ static thread_local FILOMemoryPool mainMemoryPool_{mainChunkSize};
 // if the current thread is a worker thread, use this pool
 static thread_local FILOMemoryPool threadMemoryPool_{threadlocalChunkSize};
 
-extern "C" void *gcAlignedMalloc(size_t sz) noexcept {
+#if defined _WIN32 || defined __CYGWIN__
```
Comment (Menooker) [inline lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp:242]: Maybe we should move this macro to a shared header in `CPURuntime` folder, as many cpp files will need it?

Comment (niuxiaog) [reply lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp:242]: Sure.

**Comment by Menooker** [comment]
Suggest to co-work with @huanghaixin008 . The brgemm library also needs to export the symbols.

**Comment by huanghaixin008** [comment]
> Suggest to co-work with @huanghaixin008 . The brgemm library also needs to export the symbols.

@niuxiaog  I will push my changes to this branch

File: include/gc/ExecutionEngine/CPURuntime/Utils.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//===-- Utils.h - Utils for CPURuntime --------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#if defined _WIN32 || defined __CYGWIN__
```
Comment (Menooker) [inline include/gc/ExecutionEngine/CPURuntime/Utils.h:9]: Please add the include guard for this header. :)

File: lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp (hunk 1 above)
Comment (AndreyPavlenko) [reply lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp]: May be move it even higher? I believe it's required not only for CPURuntime. What about renaming gc_version.h to gc.h and putting the common staff there?

Comment (niuxiaog) [reply lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp]: > May be move it even higher? I believe it's required not only for CPURuntime.

What about to do this only when there is a real requirement?

> renaming gc_version.h to gc.h and putting the common staff there

The name `gc_version.h` conveys more specific information and may be a better name. The newly added `CPURuntime/Utils.h` and `gc_version.h` contains macros with totally different purpose. Currently these is no file which needs to include them both. So putting them together may not be necessary.

Comment (AndreyPavlenko) [reply lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp]: > What about to do this only when there is a real requirement?

Currently, except of CPURuntime, we need it at least for lib/gc/ExecutionEngine/OpenCLRuntime/OpenCLRuntimeWrappers.cpp and src/dnnl/dnnl_graph_compiler.cpp

File: lib/gc/ExecutionEngine/OpenCLRuntime/OpenCLRuntimeWrappers.cpp [hunk 3]
```diff
@@ -18,11 +18,9 @@
 #include <stdexcept>
 #include <vector>
 
-#ifdef _WIN32
-#define OCL_RUNTIME_EXPORT __declspec(dllexport)
-#else
-#define OCL_RUNTIME_EXPORT
-#endif // _WIN32
+#include "gc_utils.h"
```
Comment (ciyongch) [inline lib/gc/ExecutionEngine/OpenCLRuntime/OpenCLRuntimeWrappers.cpp:21]: Need to update the CMakeList.txt to add the directory of newly added "gc_utils.h" to resolve the build error.

Comment (niuxiaog) [reply lib/gc/ExecutionEngine/OpenCLRuntime/OpenCLRuntimeWrappers.cpp:21]: Yes. Thanks for the advice! I will come back to this PR soon.

File: include/gc_utils.h
Comment (AndreyPavlenko) [inline include/gc_utils.h:1]: I propose moving this file into the gc folder, in order to have all the headers under the single root, and renaming to Utils.h, to be consistent with the naming style of all other headers.

File: lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt [hunk 4]
```diff
@@ -16,6 +16,7 @@ target_compile_options (GcOpenclRuntime PUBLIC -fexceptions -frtti)
 target_include_directories(GcOpenclRuntime PRIVATE
     ${MLIR_INCLUDE_DIRS}
     ${OpenCL_INCLUDE_DIRS}
+    ${PROJECT_SOURCE_DIR}/include
```
Comment (AndreyPavlenko) [inline lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt:19]: I think, you can just add GcInterface to target_link_libraries(). It should propagate all the common includes, build options, etc.

Comment (niuxiaog) [reply lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt:19]: Yes, that's also a reasonable way.

**Comment by ciyongch** [comment]
Please rebase the PR and address the comments, then we can merge it for OV integration.

File: lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt (hunk 4 above)
Comment (ciyongch) [reply lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt:19]: Do you plan to change this?

Comment (niuxiaog) [reply lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt:19]: > Do you plan to change this?

Which way do you think is better?

Comment (Menooker) [reply lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt:19]: Isn't `${PROJECT_SOURCE_DIR}/include` already in include path?

Comment (niuxiaog) [reply lib/gc/ExecutionEngine/OpenCLRuntime/CMakeLists.txt:19]: The `${PROJECT_SOURCE_DIR}/include` is not in include path, which causes the including error of `gc_utils.h`. But now `gc_utils.h` is moved into `/include/gc`, I will modify this now. Thanks for the reminder!

**Review by ciyongch — APPROVED** [review]

**Review by kurapov-peter — APPROVED** [review]
LGTM

**Pull request merged**